src/Hydration/PublicPropertyManager.php

```diff
@@ -9,13 +9,13 @@ class PublicPropertyManager
 {
     public $properties;
 
-    public function register($class)
+    public function register($class, $resolver)
     {
-        if (! $class instanceof Wireable) {
+        if (! $resolver instanceof Wireable) {
             throw new CannotRegisterPublicPropertyWithoutImplementingWireableException(get_class($class));
```

Comment:
`throw_if`?

Reply:
Yep, that would be even nicer.

Reply:
This is pretty old and already refactored.

Reply:
I think it's the way you do like it Erin :-)

https://github.com/livewire/livewire/pull/4154/files#diff-f5ec0a6e40cd9166462c2bc3c7c033aeab3ad2486b87901108c9ec202ab7f004R19